### PR TITLE
Feature/refactor

### PR DESCRIPTION
**Pull request by Priyanshu-gkm**
- renamed variables
- segregated models, serializers, views and helpers
- changed routes and endpoints
- renamed files and folders

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
 .envrc
 .DS_Store
-.pytest_cache
\ No newline at end of file
+.pytest_cache
+migrations
```
Comment (AditSoni) [inline .gitignore]: why are migrations in git ignore?

Comment (Priyanshu-gkm) [reply .gitignore]: removed migrations from gitignore and added the migrations in the latest commit

**Review by AditSoni — APPROVED** [review]

**Comment by Priyanshu-gkm** [comment]
- changed the response object body to get a standard response
- changed some variables
- changed endpoint DELETE /follow-requests/<id> to DELETE /connections/<id>
- changed some response codes
- removed relative imports with absolute imports

NOTE : test cases were made in a hurry, they are not made according to the standards. Some functions are tested within single test case due to the sequence of events.

**Review by AditSoni — APPROVED** [review]

**Comment by Priyanshu-gkm** [comment]
- modified permissions 
- refactored user deletion view
- added a new test case file for user authentication - register, login and logout 
- modified test_user.py to contain only test cases related to user operations.

**Comment by Priyanshu-gkm** [comment]
- modified test cases for post_type

**Comment by Priyanshu-gkm** [comment]
- modified post view , only authenticated users can view the posts
- added more test cases for the /posts endpoint

**Comment by Priyanshu-gkm** [comment]
- added new test file for testing connections
- modified response object for responding to follow request

**Comment by Priyanshu-gkm** [comment]
- created a new file for testing /feed endpoint
- modified feed view

**Comment by Priyanshu-gkm** [comment]
- modified view of notification for redundant code
- added a new test file for testing notifications
- renamed TestConnection -> TestFeed in test_feed.py ( missed in previous commit )

**Comment by Priyanshu-gkm** [comment]
- modified follow request model init function 
- modified follow request view, removed redundant data
- modified follow request test cases, proper segmentation and assertions added

**Review by AditSoni — APPROVED** [review]

**Pull request merged**